docs/console/console-extensibility/registration.md

```diff
@@ -0,0 +1,232 @@
+
+### How to configure correctly the extension menu item?
+
+![registration example main route](./images/registrationExampleMainRoute.png)
+
+![registration example category route](./images/registrationExampleCategoryRoute.png)
+
+The routes should contain the info to render the extension menu item on the Console sidebar, that  allow to access the extension. A route can be registered leaving `renderType` unsetted to be rendered as simple menu item or it can be a `category` route corresponding to a new custom menu group where attach menu items.
+
+:::info
+Each extension should have at least a route (that is not `category`) to have sense and could also add at most another `category` type route to add a custom group menu item
+:::
+
+A route, therefore, requires to provide the following information:
+- `id` (_required_): assign a unique id that can be used for applying overrides with activation or for attaching other menu items if the route renderType is `category`
+- `locationId` (_required_): choose a location to place your route (see the [supported locations](./locations.md))
+- `renderType`: select `category` if you want to add a new menu group, otherwise leave it unsetted to add a menu item
+- `parentId`: insert the route `id` of a menu group where you want to attach the menu item. It is possible to choose existing parentIds documented on the [supported locations page](./locations.md) or add a route of type `category` and use its `id`. This value can be omitted for the `category` route.
+- `destinationPath` (_required_): indicate the destination suffix to which you will be redirected when clicking the menu item and it will compose the URL where the iframe will be mounted; in fact, the resulting URL will be composed accoring this pattern `<locationPath>/extensions/<extensionId><destinationPath>`. This information should be defined but is ignored for the `category` route.
+- `icon`: select an icon for the new menu item and fill the `icon.name` field. You can find the icons at this [link](https://react-icons.github.io/react-icons/search/), but only Ant, Feather and Phosphor icons are supported. This value can be omitted for the `category` route.
+- `labelIntl` (_required_): insert the label to be used on the menu item or menu group and fill this field providing an object JSON `{"en": string, "it": string}` so that the text is also internationalized.
+<!-- TODO: Should be added some mentions about the order of menu items? -->
+
+#### Register Backoffice Extension Example
```

Comment:
should this be moved in a tutorial page?

Reply:
In my opinion this doc pages can be already used as tutorial, because the theorical concepts are not a lot but it is required some practical example near to it to understand them

Reply:
ok but this is not a tutorial, as it is not clearly stated that the user is following a tutorial. Therefore it's just a doc page with many examples. 

If we want this to be a tutorial let's make one and separate what is documentation from what is a sequence of well defined and exemplified steps the user can follow to fiddle around with the feature.